lib/index.js

```diff
@@ -162,6 +164,31 @@ FSTree.prototype.calculatePatch = function(otherFSTree, isEqual) {
   return removals.reverse().concat(additions);
 };
 
+FSTree.applyPatch = function(source, output, patch) {
```

Comment:
Oh I see, I assumed that is what we where doing. 

I've always thought maybe a visitor style pattern here might do the trick. Although never figure out what was a good fit.

>  Thoughts? 

I'll think on it some in the next few hours.